### PR TITLE
Show appropriate spaceship error message when having insufficient permissions

### PR DESCRIPTION
**Pull request by KrauseFx**
screenshot 2017-01-25 19 43 26

### CODE REVIEW

**Comment by KrauseFx** [comment]
I'm adding some more things here now

**Review by mfurtak — APPROVED** [review]
```ruby
caller[0][/`.*'/][1..-2]
```

is pretty crazy, but unless you can think of a way to do this that is easier to understand... I say 👍

**Comment by snatchev** [comment]
@mfurtak @KrauseFx 
```ruby
caller_locations(1,1).first.label
```

File: spaceship/lib/spaceship/client.rb [hunk 1]
```diff
@@ -50,6 +50,20 @@ class InvalidUserCredentialsError < BasicPreferredInfoError; end
...
+    class InsufficientPermissions < BasicPreferredInfoError
+      TITLE = 'Insufficient permissions for your Apple ID:'.freeze
+
+      def preferred_error_info
+        message ? [TITLE, message] : nil
+      end
+
+      # We don't want to show similar GitHub issues, as the error message
+      # should be pretty clear
+      def show_github_issues
```
Comment (lacostej) [inline spaceship/lib/spaceship/client.rb]: this is cool. Maybe should be added to a whole bunch of other errors.

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 2]
```diff
@@ -214,10 +214,17 @@ def handle_itc_response(raw)
...
+        # Sample `error` content: [["Forbidden"]]
         if errors.count == 1 and errors.first == "You haven't made any changes."
           # This is a special error which we really don't care about
         elsif errors.count == 1 and errors.first.include?("try again later")
           raise ITunesConnectTemporaryError.new, errors.first
+        elsif errors.count == 1 and errors.first.include?("Forbidden")
+          # get the method name of the request that failed
+          # `block in` is used very often for requests when surrounded for paging or retrying blocks
+          # The ! is part of some methods when they modify or delete a resource, so we don't want to show it
+          calling_method_name = caller_locations(1, 1).first.label.delete("block in").delete("!").strip
```
Comment (lacostej) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: given that we duplicate this code, consider moving it to a helper method.

**Comment by KrauseFx** [comment]
@lacostej Yeah, updated and unified the code. Agreed re GitHub issue. If you notice something, please PR on it 👍

**Comment by KrauseFx** [comment]
This is so beautiful 👍 

screenshot 2017-01-26 12 31 32
screenshot 2017-01-26 14 52 16

**Comment by KrauseFx** [comment]
I want to add some tests for this before merging

**Comment by hjanuschka** [comment]
that is so beautiful 😎

**Pull request merged**